packages/alfa-rules/src/sia-r56/rule.ts

```diff
@@ -66,7 +66,7 @@ export default Rule.Atomic.of<Page, Group<Element>>({
 
 export namespace Outcomes {
   export const differentNames = (role: Role.Name) =>
-    Ok.of(Diagnostic.of(`No two \`${role}\` have the same name.`));
+    Ok.of(SameNames.of(`No two \`${role}\` have the same name.`, role, []));
```

Comment:
I think it's just easier to keep the same structure in both case, even if the passed ones are always going to have an empty list of errors.